src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1266,14 +1255,7 @@ private Type EmitFunctionCallNode(AssociativeNode node)
                     foreach (var arg in args)
                     {
                         Type t = DfsTraverse(arg);
-                        if (isStaticOrCtor)
-                        {
-                            t = EmitCoercionCode(arg, t, parameters[index]);
-                        }
-                        else if (index > 0)
-                        {
-                            t = EmitCoercionCode(arg, t, parameters[index - 1]);
-                        }
```

Comment:
Why?

Reply:
In my tests I found that the number of parameters and arguments were always the same even in the case of static methods.

This caused a test failure, unfortunately I cannot remember which one, but if you revert this change I believe you'll see one.

Reply:
But we're doing the same over here: https://github.com/DynamoDS/Dynamo/blob/2d53bb411af05a4921a98fcba4bf9f18c960daf0/src/Engine/EmitMSIL/CodeGenIL.cs#L1180